### ISSUE TITLE
Support to pass arguments to command-based variables

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#38327]
For launch.json and task.json configuration files VS Code supports variables that are based on commands. One limitation of this mechanism is that it is not possible to pass additional arguments to the command.

Examples:
- #32726: asks for being able to pass a filter pattern to the process picker command, e.g. something like ${command:autoPickProcessWithPattern:TopTen-Server.dll}
- #9297: asks for a command that uses QuickPick to select from a list of enums, e.g. something like ${command:enumQuickPick:'choice 1','choice 2', 'choice 3'}"

**Comment by bitjson** [comment]
I'm hoping to implement https://github.com/Microsoft/vscode/pull/50758 as an extension – any news on this feature request?

**Comment by isidorn** [comment]
@weinand should we close this?

**Comment by bitjson** [comment]
@isidorn is this functionality available somehow? I'd love to be able to provide a parameter to a contributed command (for use in `launch.json`). 

It doesn't seem like there's even a good hack to make that happen though – even if I parse the configuration to consume the parameter in a different way, the user is shown an error on unrecognized configuration options. So the only way to provide information to a command is by somehow embedding it in existing configuration options.

**Comment by RobertDeRose** [comment]
Would love for this feature to be supported

**Comment by weinand** [comment]
Next week we will support passing arguments to command based variables via #64362:
```ts
{
	"version": "0.2.0",
	"configurations": [
		{
			"type": "node",
			"request": "launch",
			"name": "Launch Program",
			"program": "${input:askForXYZ}"
		}
	],
	"inputs": [
		{
			"id": "askForXYZ",
			"type": "command",
			"description": "The XYZ to use"
			"command": "extension.foobar.someCommand",
		}
	]
}
```

**Comment by busticated** [comment]
> I don't understand your question

if i'm providing tasks configuration via `vscode.workspace.registerTaskProvider()` (and related), will i be able to use "input" variables?

**Comment by weinand** [comment]
Are you able to use other variables, e.g. `${workspaceFolder}` or `${env:HOME}`?
If the answer is yes, then you are able to use "input" variables.

Sorry, I cannot give you a better answer because  this issue is "debug" related, and I'm the expert on "tasks".

**Comment by weinand** [comment]
this feature has become available via #64362.
